src/main/groovy/com/bugsnag/android/gradle/BugsnagPlugin.groovy

```diff
@@ -73,13 +82,26 @@ class BugsnagPlugin implements Plugin<Project> {
                 manifestTask.onlyIf { it.shouldRun() }
 
                 // Create a Bugsnag task to upload proguard mapping file
-                BugsnagUploadTask uploadTask = project.tasks.create("uploadBugsnag${variantName}Mapping", BugsnagUploadTask)
+                BugsnagUploadProguardTask uploadTask = project.tasks.create("uploadBugsnag${variantName}Mapping", BugsnagUploadProguardTask)
                 uploadTask.group = GROUP_NAME
                 uploadTask.manifestPath = manifestPath
                 uploadTask.applicationId = variant.applicationId
                 uploadTask.mappingFile = variant.getMappingFile()
                 uploadTask.mustRunAfter variantOutput.packageApplication
 
+                // Create a Bugsnag task to upload NDK mapping file(s)
+                BugsnagUploadNdkTask uploadNdkTask = project.tasks.create("uploadBugsnagNdk${variantName}Mapping", BugsnagUploadNdkTask)
+                uploadNdkTask.group = GROUP_NAME
+                uploadNdkTask.manifestPath = manifestPath
+                uploadNdkTask.applicationId = variant.applicationId
+                uploadNdkTask.intermediatePath = intermediatePath
+                uploadNdkTask.symbolPath = symbolPath
+                uploadNdkTask.variantName = variant.name
+                uploadNdkTask.projectDir = project.projectDir
+                uploadNdkTask.rootDir = project.rootDir
+                uploadNdkTask.toolchain = getCmakeToolchain(project, variant)
+                uploadNdkTask.mustRunAfter variantOutput.packageApplication
```

Comment:
Since this task only applies to some consumers, is there an `onlyIf` condition we could use here to determine if its appropriate before running?